server/context.go

```diff
@@ -204,6 +211,18 @@ func (ctx *Context) parseGossipBootstrapResolvers() ([]gossip.Resolver, error) {
 	return bootstrapResolvers, nil
 }
 
+// GetHTTPClient returns the context http client, initializing it
+// if needed. It uses the context Certs.
+// TODO(marc): are there odd cases where may need a mutex?
```

Comment:
might be cheaper to use https://golang.org/pkg/sync/#Once.Do here since the typical overhead is less than with a mutex